simple_parsing/replace_selections.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import copy
+import dataclasses
+import logging
+from typing import Any, overload
+
+from simple_parsing.annotation_utils.get_field_annotations import (
+    get_field_type_from_annotations,
+)
+from simple_parsing.helpers.subgroups import Key
+from simple_parsing.replace import replace
+from simple_parsing.utils import (
+    DataclassT,
+    contains_dataclass_type_arg,
+    is_dataclass_instance,
+    is_dataclass_type,
+    is_optional,
+    is_union,
+    unflatten_keyword,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@overload
+def replace_selections(
+    obj: DataclassT,
+    changes_dict: dict[str, Any],
+    selections: dict[str, Key | DataclassT] | None = None,
+) -> DataclassT:
+    ...
+
+
+@overload
+def replace_selections(
```

Comment:
Do you have a type checker enabled, out of curiosity? (e.g. Pylance, mypy, etc)
You should be getting an error here because `@overload` should be directly above the overloaded function.
Here they are above `replace_selected_dataclass`.

Reply:
I use only Pylance for type checking. I wonder why it didn't report any error for this. May I just overlooked any error or warning for this.